dockers/texcmp/texcmp.js

```diff
@@ -0,0 +1,244 @@
+        var uw = Math.max(browser.width, latex.width) + Math.abs(ox);
+        var uh = Math.max(browser.height, latex.height) + Math.abs(oy);
+        var bx = Math.max(ox, 0);
+        var by = Math.max(oy, 0);
+        var lx = Math.max(-ox, 0);
+        var ly = Math.max(-oy, 0);
+        return execFile("convert", [
+            "(", pngFile, "-grayscale", "Rec709Luminance",
+            "-extent", uw + "x" + uh + "-" + lx + "-" + ly,
+            ")",
+            "(", browserFile, "-grayscale", "Rec709Luminance",
+            "-extent", uw + "x" + uh + "-" + bx + "-" + by,
+            ")",
+            "(", "-clone", "0-1", "-compose", "darken", "-composite", ")",
+            "-channel", "RGB", "-combine", "-trim", diffFile
+        ]);
+    }).then(function() {
+        console.log("Compared " + key);
+    });
+};
+
+// Create a directory, but ignore error if the directory already exists.
+function mkdir(dir) {
+    return Q.nfcall(fs.mkdir, dir).catch(function(err) {
+        if (err.code !== "EEXIST")
+            throw err;
+    });
+}
+
+// Execute a given command, and return a promise to its output.
+function execFile(cmd, args, opts) {
+    var deferred = Q.defer();
+    child_process.execFile(cmd, args, opts, function(err, stdout, stderr) {
+        if (err) {
+            console.error("Error executing " + cmd + " " + args.join(" "));
+            console.error(stdout + stderr);
+            err.stdout = stdout;
+            err.stderr = stderr;
+            deferred.reject(err);
+        } else {
+            deferred.resolve(stdout);
+        }
+    });
+    return deferred.promise;
+}
+
+// Read given file, and apply FFT transformation to it.
+function fftPNG(file) {
+    var deferred = Q.defer();
+    var onerror = deferred.reject.bind(deferred);
+    var stream = fs.createReadStream(file);
+    stream.on("error", onerror);
+    pngparse.parseStream(stream, parsed);
+    return deferred.promise;
+
+    function parsed(err, image) {
+        if (err) return onerror(err);
+        var real = newMatrix();
+        var imag = newMatrix();
+        var idx = 0;
+        var nchan = image.channels;
+        var alphachan = 1 - (nchan & 1);
+        var colorchan = nchan - alphachan;
+        for (var y = 0; y < image.height; ++y) {
+            for (var x = 0; x < image.width; ++x) {
+                var c;
+                var v = 0;
+                for (c = 0; c < colorchan; ++c) {
+                    v += 255 - image.data[idx++];
+                }
+                for (c = 0; c < alphachan; ++c) {
+                    v += image.data[idx++];
+                }
+                real.set(y, x, v);
+            }
+        }
+        fft(1, real, imag);
+        deferred.resolve({
+            real: real,
+            imag: imag,
+            width: image.width,
+            height: image.height
+        });
+    }
+}
+
+// Create a new matrix of preconfigured dimensions, initialized to zero
+function newMatrix() {
```

Comment:
To me, it feels conceptually close, so this was intentional. But I have no strong opinion there, `createMatrix` is fine with me, too.